### PR TITLE
Feat: set_hostname - Add fqdn capability

### PR DESCRIPTION
**Pull request by oxedions**
Add FQDN capability for set_hostname.

Request made for an HPC cluster. Seems reasonable.

### CODE REVIEW

**Comment by hmeAtos** [comment]
ok for me, good feature  @oxedions

**Review by hmeAtos — COMMENTED** [review]
have you write a documentation for set `set_hostname_fully_qualified_domaine_name` parameter ?

**Comment by oxedions** [comment]
Good point :-)
On it

File: roles/core/set_hostname/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+---
+set_hostname_fully_qualified_domaine_name: false
```
Comment (btravouillon) [inline roles/core/set_hostname/defaults/main.yml:2]: Could you define a shorter parameter name? `set_hostname_fqdn`?

File: roles/core/set_hostname/molecule/default/converge.yml [hunk 2]
```diff
@@ -1,6 +1,8 @@
 ---
 - name: Converge
   hosts: all
+  vars:
+    domain_name: bluebanquise.local
```
Comment (btravouillon) [inline roles/core/set_hostname/molecule/default/converge.yml:5]: Is `domain_name` a mandatory variable?

File: roles/core/set_hostname/readme.rst [hunk 3]
```diff
@@ -14,6 +14,7 @@ NA.
 Changelog
 ^^^^^^^^^
 
+* 1.0.3: Add fully qualified domaine name capability. Benoit Leveugle <benoit.leveugle@gmail.com>
```
Comment (btravouillon) [inline roles/core/set_hostname/readme.rst:17]: s/domaine/domain/

**Review by btravouillon — CHANGES_REQUESTED** [review]

File: roles/core/set_hostname/defaults/main.yml (hunk 1 above)
Comment (oxedions) [reply roles/core/set_hostname/defaults/main.yml]: Done 🙂

File: roles/core/set_hostname/molecule/default/converge.yml (hunk 2 above)
Comment (oxedions) [reply roles/core/set_hostname/molecule/default/converge.yml:5]: Yes, due to ternary 😐 Ansible always check all variables inside the ternary, even if not used.

File: roles/core/set_hostname/readme.rst (hunk 3 above)
Comment (oxedions) [reply roles/core/set_hostname/readme.rst]: Good catch, done.

File: roles/core/set_hostname/molecule/default/converge.yml (hunk 2 above)
Comment (btravouillon) [reply roles/core/set_hostname/molecule/default/converge.yml:5]: This needs to be documented then.

**Review by btravouillon — CHANGES_REQUESTED** [review]
With this change the parameter `domain_name` must be defined. This needs to be documented somewhere.

**Review by btravouillon — APPROVED** [review]
Looks good to me. 👍

**Comment by oxedions** [comment]
Thanks for reviewing.

**Pull request merged**